### DISCUSSION TITLE
Poetry on PubPub

### DISCUSSION THREAD

**Discussion by writingpoli** [knowledgefutures/pubpub#1493] [Support]
SAR gets lots of poetry submissions. 

Lots of poets like to use strange formatting. 

**Tips/tricks/ideas to enhance PubPub's capabilities regarding spacing and the like for experimental poetry?**

**Comment by gabestein** [comment]
If you have some examples, that would be helpful to know what we're dealing with! Otherwise, a few scattered thoughts, though I bet others have better suggestions:

- Little-known feature from our end: using shift-return/enter, you can do a single, rather than double, line-break. 
- PubPub now supports left, right, and center alignment on any given line (including within lists and tables).
- We have full unicode character support, including emojis, etc. There are some funky unicode characters that allow you to do different spacing.

**Reply by writingpoli** [reply]
Beyond the teaching folks the line break method, our biggest struggle is with spacing. I'm certain alignment options will help. Some poets like to space their lines irregularly -- hanging indents, extra spaces, etc. Some affordance for those would be helpful, I think. (Especially if we can maintain them in the auto-generated outputs, too. I like to minimize my future print edition editing where I can.)

**Reply by gabestein** [reply]
Ah! I wonder if we could support various forms of indentation...or even just run-of-the-mill indentation. @idreyn or @isTravis, penny for your thoughts there.

**Reply by writingpoli** [reply]
Usually, I'd take the hard route and input the necessary special characters in the HTML view of other CMS editors.

**Reply by isTravis** [reply]
The first thing that jumped to mind was simply adding in a helper UI for inserting special HTML characters to manage spacing (I'm picturing a more simple, modern version of Word's Special Character Palette). Even with that though, I could imagine some formatting where that would still be insufficient.

A real design challenge will be setting expectations for different devices. Should lines, font-sizes, etc remain the same across mobile, desktop, and print even if it comes at the cost of reduced legibility (e.g. on narrow mobile devices)?

I think providing robust line-by-line margin customization a la Word or Google docs won't be possible because we don't assume a fixed page width as they do. 

I wonder how far we could get with a Code-Block-like node that didn't use monospace font (e.g. it looked "normal"). Code-blocks use `<pre>` tags which maintain whitespace and don't auto-flow text onto new lines.

**Reply by writingpoli** [reply]
I actually like your first idea. I'd really like if there were a small media library we could utilize for recurring elements, too. Having to find and upload images of elements for each Pub is a hassle. I started doing it at the bottom of our pubs briefly but can't get others to remember to use them.

**Reply by writingpoli** [reply]
If we could use HTML on the Pub editor, that'd suffice.

**Comment by writingpoli** [comment]
Here's our current solution, which doesn't help accessibility: https://www.sareview.org/pub/kfzidmr7/
